### DISCUSSION TITLE
Thread Safe

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2931] [Help]
Im using MagickWand method to reisze the Image 

NewMagickWand()
MagickReadImage()
MagickResizeImage()
MagickWriteImages()
DestroyMagickWand()

These method are thread safe , will support multi-threading ?

**Comment by urban-warrior** [comment]
The ImageMagick API's are thread safe, including MagickCore, MagickWand, and Magick++.

**Reply by programmerarunkumar** [reply]
Can you see this post #2925

MagickReadImage(MagickWand *wand,const char *filename) return MagickFalse 

I haved check the image before using MagickReadImage() , it exist & it is valid image , but it return MagickFalse

**Comment by urban-warrior** [comment]
If a method fails, recommended practice is to review the associated exception, for example:

```
#define ThrowAPIException(wand) \
{ \
  description=MagickGetException(wand,&severity); \
  (void) FormatLocaleFile(stderr,"%s %s %lu %s\n",GetMagickModule(), \
    description); \
  description=(char *) MagickRelinquishMemory(description); \
  exit(-1); \
}
...
status=MagickReadImage(magick_wand,path);
if (status == MagickFalse)
  ThrowAPIException(magick_wand);
```
What does the exception tell us about why it failed to read the image?
